### DISCUSSION TITLE
Course listing visible only to enrolled participants

### DISCUSSION THREAD

**Discussion by RachelBrodsky** [inducer/relate#865] [Q&A]
How do we make courses visible to students after they login?

We can manually make the course public on the main page by going into the course settings and un-checking ‘Only visible to course staff’, but then the course is public to everyone (regardless of whether or not they sign in). Is there a way to only make a course visible after a student signs in?

**Comment by inducer** [comment]
Easy to do without code: Simply give the `view_hidden_course_page` permission to all the roles that are supposed to be able to see it. (Note that roles can only be edited in the admin ATM.)

See the visibility computation here:

https://github.com/inducer/relate/blob/1f18b9d1924ce54c8310d8f0c37950ba6e4064d9/course/views.py#L111-L114

**Reply by RachelBrodsky** [reply]
Thank you! Where can we change the view hidden course page permissions?

**Reply by inducer** [reply]
Click on "Staff > Admin site", then "Participation roles", then filter down to your course, pick the "participant" role, and add `view_hidden_course_page`.
